### ISSUE TITLE
 logit = self.fc(fea) * self.s

### ISSUE THREAD

**Issue by xaioduzi** [alipay/cvpr2020-plant-pathology#4]
Hello,I want to ask a question.
In the models.py,
class BinaryHead(nn.Module):
    def __init__(self, num_class=4, emb_size=2048, s=16.0):
        super(BinaryHead, self).__init__()
        self.s = s
        self.fc = nn.Sequential(nn.Linear(emb_size, num_class))

    def forward(self, fea):
        fea = l2_norm(fea)
        logit = self.fc(fea) * self.s
        return logit

why does the output of the linear unit need to multify 16?
I am very puzzled.
Thanks!

**Comment by nickyi1990** [comment]
this class (BinaryHead) has never been used, u can ignore it

**Comment by xaioduzi** [comment]
> this class (BinaryHead) has never been used, u can ignore it

In the code of the github,this class is used.
Is 16 designed as a hyperparameter or otherwisre?

class BinaryHead(nn.Module):
    def __init__(self, num_class=2, emb_size=2048, s=16.0):
        super(BinaryHead, self).__init__()
        self.s = s
        self.fc = nn.Sequential(nn.Linear(emb_size, num_class))

    def forward(self, fea):
        fea = l2_norm(fea)
        logit = self.fc(fea) * self.s
        return logit


class se_resnext50_32x4d(nn.Module):
    def __init__(self):
        super(se_resnext50_32x4d, self).__init__()

        self.model_ft = nn.Sequential(
            *list(pretrainedmodels_se_resnext50_32x4d(num_classes=1000, pretrained="imagenet").children())[
                :-2
            ]
        )
        self.avg_pool = nn.AdaptiveAvgPool2d((1, 1))
        self.model_ft.last_linear = None
        self.fea_bn = nn.BatchNorm1d(2048)
        self.fea_bn.bias.requires_grad_(False)
        self.binary_head = BinaryHead(2, emb_size=2048, s=1)
        self.dropout = nn.Dropout(p=0.2)

    def forward(self, x):

        img_feature = self.model_ft(x)
        img_feature = self.avg_pool(img_feature)
        img_feature = img_feature.view(img_feature.size(0), -1)
        fea = self.fea_bn(img_feature)
        # fea = self.dropout(fea)
        output = self.binary_head(fea)

        return output

**Comment by nickyi1990** [comment]
> > this class (BinaryHead) has never been used, u can ignore it
> 
> In the code of the github,this class is used.
> Is 16 designed as a hyperparameter or otherwisre?
> 
> class BinaryHead(nn.Module):
> def **init**(self, num_class=2, emb_size=2048, s=16.0):
> super(BinaryHead, self).**init**()
> self.s = s
> self.fc = nn.Sequential(nn.Linear(emb_size, num_class))
> 
> ```
> def forward(self, fea):
>     fea = l2_norm(fea)
>     logit = self.fc(fea) * self.s
>     return logit
> ```
> 
> class se_resnext50_32x4d(nn.Module):
> def **init**(self):
> super(se_resnext50_32x4d, self).**init**()
> 
> ```
>     self.model_ft = nn.Sequential(
>         *list(pretrainedmodels_se_resnext50_32x4d(num_classes=1000, pretrained="imagenet").children())[
>             :-2
>         ]
>     )
>     self.avg_pool = nn.AdaptiveAvgPool2d((1, 1))
>     self.model_ft.last_linear = None
>     self.fea_bn = nn.BatchNorm1d(2048)
>     self.fea_bn.bias.requires_grad_(False)
>     self.binary_head = BinaryHead(2, emb_size=2048, s=1)
>     self.dropout = nn.Dropout(p=0.2)
> 
> def forward(self, x):
> 
>     img_feature = self.model_ft(x)
>     img_feature = self.avg_pool(img_feature)
>     img_feature = img_feature.view(img_feature.size(0), -1)
>     fea = self.fea_bn(img_feature)
>     # fea = self.dropout(fea)
>     output = self.binary_head(fea)
> 
>     return output
> ```

sorry for missing this code, It's just for testing, we set s=1 which is equal to normal classification head. when set to different values, it acts like regularization.
![image](https://user-images.githubusercontent.com/13381684/95339040-2d461500-08e6-11eb-95a7-4c41c36a8106.png)
